### PR TITLE
chore: adjust readme.md for beginners

### PR DESCRIPTION
**Pull request by olivierhalupczok**
Signed-off-by: Olivier Halupczok <olivier.halupczok@boldare.com>

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

Current Readme.md file was shorter and was mainly aimed for more advanced users.

Issue Number: N/A

## What is the new behavior?

N/A - Only changes to the `Readme.md` file, so it is easier to understand for beginners.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -71,38 +82,125 @@ Add the following content to the `.env` file:
...
+  # defualt command for NestJS is:
+  npm run start
+  ```
+
+  Then you can test the assistant.
+  1. First, you need to create a thread. You can create one
+  by sending a POST request to the `/assistant/threads` endpoint with the empty body:
+  ```
+  {}
+  ```
```
Comment (sebastianmusial) [inline README.md:167]: empty body - this information should be enough, we don't have to pass the empty object in the body

File: README.md [hunk 2]
```diff
@@ -71,38 +82,125 @@ Add the following content to the `.env` file:
...
+
+### Step 5: Running the application and testing
+
+Run your application and this will allow you to test the assistant. 
+  
+  ```bash
+  # use this if you are using the repository:
+  npm run start:dev
+
+  # if you are using NestJS please check the npm scripts in the package.json file
```
Comment (sebastianmusial) [inline README.md:157]: `if you are using your own NestJS application, please check the npm scripts in the package.json file`

File: README.md [hunk 3]
```diff
@@ -71,38 +82,125 @@ Add the following content to the `.env` file:
...
+  temperature: 0.05,
+};
+
+// Additional configuration for our assistant
+export const assistantConfig: AssistantConfigParams = {
+  id: process.env['ASSISTANT_ID'],
+  params: assistantParams,
+  filesDir: './apps/api/src/app/knowledge',
+  toolResources: {
+    codeInterpreter: {
```
Comment (sebastianmusial) [inline README.md:117]: `file_search` should be better for readme because it seems to be more popular.

File: README.md [hunk 4]
```diff
@@ -71,38 +82,125 @@ Add the following content to the `.env` file:
...
+// chat.config.ts file
+import { AssistantConfigParams } from '@boldare/openai-assistant';
+import { AssistantCreateParams } from 'openai/resources/beta';
+
+// Default OpenAI configuration
+export const assistantParams: AssistantCreateParams = {
+  name: 'Your assistant name',
+  instructions: `You are a chatbot assistant. Speak briefly and clearly.`,
+  tools: [
+    { type: 'code_interpreter' },
```
Comment (sebastianmusial) [inline README.md:105]: `file_search` should be better for readme because it seems to be more popular.

File: README.md [hunk 5]
```diff
@@ -71,38 +82,125 @@ Add the following content to the `.env` file:
...
+  id: process.env['ASSISTANT_ID'],
+  params: assistantParams,
+  filesDir: './apps/api/src/app/knowledge',
+  toolResources: {
+    codeInterpreter: {
+      fileNames: [],
+    },
+  },
+};
+```
```
Comment (sebastianmusial) [inline README.md:122]: I am not sure if we want to display here the configuration or link to the example would be enough. 🤔

Comment (olivierhalupczok) [reply README.md:122]: Unless we provide a code snippet, the reader(especially non technical one) may encounter a friction point. I perceive this as an opportunity to make it more popular throughout the non-technical persona users. This is also why I tried to make it as short as possible and threw the file search out as this takes a few lines.

File: README.md [hunk 6]
```diff
@@ -71,38 +82,126 @@ Add the following content to the `.env` file:
...
+  params: assistantParams,
+  filesDir: './apps/api/src/app/knowledge',
+  toolResources: {
+    file_search: {
+      // Provide files if you use file_search tool
+      files: [
+        {
+          name: 'example',
+          content: 'This is an example file.',
+        },
```
Comment (sebastianmusial) [inline README.md:121]: Please fix the structure:
```
 fileSearch: {
    fileNames: ['this-is-example-file-1.md', 'this-is-example-file-2.md'],
  },
```

**Review by sebastianmusial — APPROVED** [review]

**Pull request merged**